src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -1093,6 +1094,48 @@ class spell_warr_sword_and_board : public SpellScriptLoader
         }
 };
 
+// 215538 - Trauma
+class spell_warr_trauma : public SpellScriptLoader
+{
+public:
+    spell_warr_trauma() : SpellScriptLoader("spell_warr_trauma") { }
+
+    class spell_warr_trauma_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_warr_trauma_AuraScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo
+            ({
+                SPELL_WARRIOR_TRAUMA_EFFECT
+            });
+        }
+
+        void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+        {
+            if (Unit* target = eventInfo.GetActionTarget())
+            {
+                //Get the Remaining Damage from the aura (if exist)
+                int32 remainingDamage = target->GetRemainingPeriodicAmount(target->GetGUID(), SPELL_WARRIOR_TRAUMA_EFFECT, SPELL_AURA_PERIODIC_DAMAGE);
+                //Get 25% of damage from the spell casted (Slam & Whirlwind) plus Remaining Damage from Aura
+                int32 damage = int32(CalculatePct(eventInfo.GetDamageInfo()->GetDamage(), aurEff->GetAmount()) / aurEff->GetTotalTicks()) + remainingDamage;
```

Comment:
My mistake here, I said "yes" to wrong code you suggested.
This still doesn't work because `aurEff` in `OnEffectProc` hook is the effect of the proc aura itself (215538) and this spell is not periodic (no ticks) so `aurEff->GetTotalTicks()` is 1 here. The correct function to use would be `sSpellMgr->AssertSpellInfo(SPELL_WARRIOR_TRAUMA_EFFECT)->GetMaxTicks()`